content/articles/understanding-and-working-javascript-cookies/index.md

```diff
@@ -0,0 +1,161 @@
+
+Take a look at the example below. This cookie has a 30-day expiration date.
+
+```javascript
+document.cookie = "Username = coolDesign; max-age =" + 30 * 24 * 60 * 60;
+```
+
+A valid cookie time can also be specified using the `expires` element. This functionality will not be destroyed in seconds and will take the actual date the cookie expires (in GMT/UTC format).
+
+Take a look at the example below.
+
+```javascript
+document.cookie = "Username = stunningDesign; expires =Fri, 31 jan 2022 23:59:59 GMT";
+```
+
+On all web pages, or in any subdirectory of that directory, a cookie is always available. If you specify the `path`, the cookie will be available on all web sites in that subdirectory as well. For instance, if the path is set to `/`, the cookie will be available across the whole website, regardless of the page it was created on.
+
+```javascript
+document.cookie = "Username = ExcellentDesign; Path = /";
+```
+
+If you want a cookie to work across many subdomains, use the `domain` attribute. Only the pages in the domain in which the cookie was set are accessible.
+
+```javascript
+document.cookie = "Username = ExcellentDesign; path = /; domain = sample.com";
+```
+
+Additionally, if the `secure` property is set, the cookie will only be delivered over secure (i.e. encrypted) connections, such as HTTPS.
+
+```javascript
+document.cookie =
+  "Username = ExcellentDesign; path = /; domain = sample.com; secure";
+```
+
+### Reading a cookie
+Because the `document.cookie` item provides a string comprising the semicolon and the space separated by all cookies (ie., `name = value pair`, e.g., `firstName = excellent; lastName = design;`), a cookie is more difficult to comprehend. 
+
+There are no properties defined for the cookie in this thread, such as expires, domain, path, and others. To locate each cookie in a list, partition the cookie into individual `name = value pairs` and search for a specific name using the `split()` method.
+
+```javascript
+function setCookie(name, value, daysToLive) {
+    var cookie = name + "=" + encodeURIComponent(value);
+    if(typeof daysToLive === "number") {
+        cookie += "; max-age=" + (daysToLive*24*60*60);
+        document.cookie = cookie;
+    }
+}
+function getCookie(name) {
+    var cookieArr = document.cookie.split(";");
+    for(var i = 0; i < cookieArr.length; i++) {
+        var cookiePair = cookieArr[i].split("=");
+        if(name == cookiePair[0].trim()) {
+            return decodeURIComponent(cookiePair[1]);
+        }
+    }
+    return null;
+}
+
+function checkCookie() {
+    var firstName = getCookie("firstName");
+    if(firstName != "") {
+        alert("Welcome again, " + firstName);
+    } else {
+        firstName = prompt("Please enter your first name:");
+        if(firstName != "" && firstName != null) {
+            function
+            setCookie("firstName", firstName, 30);
+        }
```

Comment:
Here, I guess you are trying to call the function `setCookie()`. Please remove the `function` keyword .